enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -1017,6 +1021,17 @@ func (c *FirecrackerContainer) LoadSnapshot(ctx context.Context) error {
 	if err != nil {
 		return status.WrapError(err, "failed to get snapshot")
 	}
+
+	// Set unique per-run identifier on the vm metadata so this exact snapshot
+	// run can be identified
+	if snap.GetVMMetadata() == nil {
+		md := &repb.VMMetadata{
+			VmId:       c.id,
+			SnapshotId: c.id,
+		}
+		snap.SetVMMetadata(md)
+	}
+	snap.GetVMMetadata().SnapshotId = c.id
```

Comment:
hm I am a little confused why we are setting the snapshot ID when unpacking the snapshot - I would think that since we've just unpacked the snapshot, the ID should stay the same until just before we're about to commit a new version to AC?